packages/workbox-precaching/_default.mjs

```diff
@@ -211,9 +211,24 @@ moduleExports.addRoute = (options) => {
     }
 
     let responsePromise = caches.open(cacheName)
-    .then((cache) => {
-      return cache.match(precachedUrl);
-    });
+      .then((cache) => {
+        return cache.match(precachedUrl);
+      }).then((cachedResponse) => {
+        if (cachedResponse) {
+          return cachedResponse;
+        }
+
+        // Fall back to the network if we don't have a cached response (perhaps
+        // due to manual cache cleanup).
+        if (process.env.NODE_ENV !== 'production') {
+          logger.debug(`The precached response for ` +
+            `${getFriendlyURL(precachedUrl)} in ${cacheName} was not found. ` +
+            `Falling back to the network instead.`);
+        }
+
+        return fetch(precachedUrl);
```

Comment:
Should we cache the file as well?

Reply:
Good question. The downside of caching is that we could end up in a state where we have a response cached that doesn't match the revision information inlined in the manifest/stored in IDB.

I don't know if that's preferable to having nothing cached at all.

One advantage to just logging rather than also attempting to "repair" the cache is that a developer who sees this message would know that their cache is in an inconsistent state. If we attempted to repair the cache but end up caching a different revision of a given URL, that won't end up logging anything the next time the same resource is requested, and the inconsistency would be hidden.

When implementing something similar in `sw-precache`, I just went with the network + logging route:

https://github.com/GoogleChromeLabs/sw-precache/blob/2ee24a96bb4812e83f47f6859361ba5e1bd535f5/service-worker.tmpl#L171-L176